### PR TITLE
Issue 849

### PR DESCRIPTION
**Pull request by martins0n**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

### CODE REVIEW

File: etna/auto/auto.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        study = self._optuna.study.get_trials()
+
+        study_params = [
+            {**trial.user_attrs, "pipeline": get_from_params(**trial.user_attrs["pipeline"]), "state": trial.state}
+            for trial in study
+        ]
+
+        return pd.DataFrame(study_params)
+
+    def top_k(self, k: int = 5) -> List[Pipeline]:
```
Comment (Mr-Geekman) [inline etna/auto/auto.py:176]: Should we add full docstring?

File: etna/auto/pool/templates.py [hunk 2]
```diff
@@ -4,7 +4,6 @@
...
     {'_target_': 'etna.pipeline.Pipeline', 'horizon': '${__aux__.horizon}', 'model': {'_target_': 'etna.models.MovingAverageModel', 'window': '${mult:${horizon},2}'}, 'transforms': [{'_target_': 'etna.transforms.TimeSeriesImputerTransform', 'in_column': 'target', 'strategy': 'mean'}, {'_target_': 'etna.transforms.LinearTrendTransform', 'in_column': 'target'}]} ,
-    {'_target_': 'etna.pipeline.Pipeline', 'horizon': '${__aux__.horizon}', 'model': {'_target_': 'etna.models.ProphetModel'}, 'transforms': [{'_target_': 'etna.transforms.TimeSeriesImputerTransform', 'in_column': 'target', 'strategy': 'forward_fill', 'window': -1}, {'_target_': 'etna.transforms.DensityOutliersTransform', 'in_column': 'target'}, {'_target_': 'etna.transforms.TimeSeriesImputerTransform', 'in_column': 'target', 'strategy': 'mean'}, {'_target_': 'etna.transforms.LinearTrendTransform', 'in_column': 'target'}]} ,
```
Comment (Mr-Geekman) [inline etna/auto/pool/templates.py:7]: What was wrong with it?

File: etna/metrics/utils.py [hunk 3]
```diff
@@ -29,3 +34,60 @@ def compute_metrics(
...
     return metrics_values
+
+
+def percentile(n: int):
+    """Percentile for pandas agg."""
+
+    def percentile_(x):
+        return np.nanpercentile(a=x.values, q=n)
+
+    percentile_.__name__ = "percentile_%s" % n
```
Comment (Mr-Geekman) [inline etna/metrics/utils.py:45]: f-strings?

File: etna/auto/auto.py [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+from etna.metrics import SMAPE
+from etna.metrics import MedAE
+from etna.metrics import Metric
+from etna.metrics import Sign
+from etna.metrics.utils import MetricAggregationStatistics
+from etna.metrics.utils import aggregate_metrics_df
+from etna.pipeline import Pipeline
+
+
+class _Callback(Protocol):
```
Comment (Mr-Geekman) [inline etna/auto/auto.py:30]: Why do we want to keep it private?

File: tests/test_auto/test_auto.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+
+@patch("etna.auto.auto.ConfigSampler", return_value=MagicMock())
+@patch("etna.auto.auto.Optuna", return_value=MagicMock())
+def test_init_optuna(
+    optuna_mock,
+    sampler_mock,
+    auto=MagicMock(),
+):
+
```
Comment (Mr-Geekman) [inline tests/test_auto/test_auto.py:84]: Empty line.

File: etna/auto/auto.py [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+    def __call__(self, metrics_df: pd.DataFrame, forecast_df: pd.DataFrame, fold_info_df: pd.DataFrame) -> None:
+        ...
+
+
+class _Initializer(Protocol):
+    def __call__(self, pipeline: Pipeline) -> None:
+        ...
+
+
+class Auto:
```
Comment (Mr-Geekman) [inline etna/auto/auto.py:40]: What about `runs_best` and `stack_result`?

Comment (Mr-Geekman) [reply etna/auto/auto.py:40]: Is it replaced with `summary`?

File: tests/test_auto/test_auto.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
+from os import unlink
```
Comment (Mr-Geekman) [inline tests/test_auto/test_auto.py:1]: Methods `top_k` and `summary` aren't tested.

File: etna/auto/auto.py (hunk 4 above)
Comment (martins0n) [reply etna/auto/auto.py:30]: Don't see any reason to make it public

File: etna/auto/auto.py (hunk 6 above)
Comment (martins0n) [reply etna/auto/auto.py:40]: `runs_result` -> `summary`
`stack_result` -> `top_k`

File: pyproject.toml [hunk 8]
```diff
@@ -247,6 +247,7 @@ filterwarnings = [
     "ignore: The 'check_less_precise' keyword in testing",
     "ignore: Feature names only support names that are all strings",
     "ignore: Given top_k=.* is less than n_segments. Algo will filter data without Gale-Shapley run.",
+    "ignore: Call to deprecated create function",
```
[the remark this answers is not in the document]
Comment (martins0n) [reply pyproject.toml:250]: I've added comments

**Review by Mr-Geekman — APPROVED** [review]
Looks fine.

**Pull request merged**